TROUBLESHOOTING.md

```diff
@@ -0,0 +1,155 @@
+To help us to troubleshoot your issues, we will need some performance insight from your application.
+
+On Linux or macOS, one of the best way to gather performance insight would be to generate CPU and allocation [FlameGraphs](https://github.com/brendangregg/FlameGraph) 
+via [Async Profiler](https://github.com/jvm-profiling-tools/async-profiler).
+
+If you want a deeper introduction to Async Profiler, go read [this article](https://hackernoon.com/profiling-java-applications-with-async-profiler-049s2790).
+
+## Installing Async Profiler
+
+To install Async Profiler, go to the [release page](https://github.com/jvm-profiling-tools/async-profiler/releases) and download the latest release.
+
+Async Profiler depends on `perf_events`. <br/>
+To allow capturing kernel call stacks using `perf_events` from a non-root process,
+you must make some OS configuration.
+
+For your session:
+
+```shell script
+echo 1 | sudo tee /proc/sys/kernel/perf_event_paranoid
+echo 0 | sudo tee /proc/sys/kernel/kptr_restrict
+```
+
+Or definitively using `sysctl`:
+
+```shell script
+sudo sysctl -w kernel.perf_event_paranoid=1
+sudo sysctl -w kernel.kptr_restrict=0
+```
+
+For allocation profiling, you also need to install HotStop debug symbol.
+
+Depending on your Linux and Java distribution this can be done via:
+
+```shell script
+# Ubuntu/Debian - Java 8
+apt install openjdk-8-dbg
+
+# Ubuntu/Debian - Java 11
+ apt install openjdk-11-dbg
+
+# On CentOS, RHEL and some other RPM-based distributions
+debuginfo-install java-1.8.0-openjdk
+```
+
+If needed, see this section in the documentation site for details: [allocation profiling](https://github.com/jvm-profiling-tools/async-profiler#allocation-profiling).
+
+## Profiling application runtime with Async Profiler
+
+Asych Profiler comes with a Java agent, and a command line.
+
+To profile application during its runtime, I recommend using the command line as you can choose when to start the profiler, and avoid your profile to be bloated with startup events. <br/>
+Indeed, when an application start, it does a lot a bootstraping instruction, that will no more occur during the rest of the run of your application. <br/>
+By starting the profiling when needed, you avoid these bootstrap instructions to be part of the profile.
+
+When you profile an application runtime, it's good to create some load on your application using a load generator tool (wrk, gatling, ...),
+and to start profiling only after some warmup time to allow the Just In Time compiler of Java to optimize your application code (and your database cache to warmup, etc...).
+
+### CPU profiling
+
+To start CPU profiling, enter the following command:
+
+```shell script
+./profiler.sh start -b 4000000 <pid>
+```
+
+`-b 4000000` is used to increase the frame buffer size as the default is usually too small.
+
+To end profiling and gathering the result you can launch the same command with the `stop` subcommand, this will tells you if the buffer frame was too small. <br/>
```

Comment:
```suggestion
To end profiling and gather the results you can launch the same command with the `stop` subcommand, this will tells you if the buffer frame was too small. <br/>
```

Reply:
s/this will tells/this will tell/